documentation/reference/component-reference.html

```diff
@@ -0,0 +1,383 @@
+---
+# Copyright Verizon Media. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+title: "Container Component Reference"
+---
+
+
+<h2 id="component-types">Component Types</h2>
+<p>
+  The container has a large number of different types of plug-ins available to a developer.
+  To start creating components, see <a href="../vespa-plugins.html">Vespa Plugins</a>.
+</p><p>
+<img src="../img/container_components.svg">
+</p><p>
+  Starting at the top of the diagram,
+  <a href="#server">server</a> instances are used for supporting external clients
+  (if no suitable implementations of
+  <code><a href="https://github.com/vespa-engine/vespa/blob/master/jdisc_core/src/main/java/com/yahoo/jdisc/service/ServerProvider.java">
+  com.yahoo.jdisc.service.ServerProvider</a></code> are available.
+  A client may or may not be a network client, so it is related to the network layer.
+  The network layer associates a Request with a <a href="#response-handlers">response handler</a>
+  and routes it to the correct type of <a href="#request-handlers">request handler</a>
+  (typically based on URI binding patterns).
+</p><p>
+  If an application needs lightweight request-response processing
+  using decomposition by a series of chained logical units,
+  the <a href="../jdisc/processing.html">processing framework</a>
+  is the correct family of components to use.
+  The request will be routed from ProcessingHandler through one or more chains of
+  <a href="#processor-instances">Processor</a> instances.
+  The exact format of the output is customizable using a <a href="#renderers">Renderer</a>.
+</p><p>
+  If doing search queries, SearchHandler will create a Query object,
+  route that to the pertinent chain of <a href="#searchers">Searcher</a> instances,
+  and associate the returned Result with the correct <a href="#renderers">Renderer</a> instance
+  for optional customization of the output format.
+</p><p>
+  The DocumentProcessingHandler is usually invoked from messagebus,
+  and used for feeding documents into an index or storage.
+  The incoming data is used to build a Document object,
+  and this is then feed through a chain of <a href="#document-processors">DocumentProcessor</a> instances.
+</p><p>
+  If building an application with custom HTTP APIs, for instance arbitrary REST APIs,
+  the easiest way is building a custom <a href="#request-handlers">RequestHandler</a>.
+  This gets the Request, which is basically a set of key-value pairs,
+  and returns a stream of arbitrary data back to the network.
+</p><p>
+  <a href="#components">Component</a> is the basic building block in the container.
+  A component may be injected into other components,
+  including all the different request handlers, document processors and so on mentioned above,
+  and will get a destructor like method invoked when no other component is using it any more.
+</p>
+
+
+<table class="table">
+  <tr id="server">
+    <th>Server</th>
```

Comment:
FYI the documentation for `ServerProvider` is no longer relevant as jdisc standalone/core is no more.

Reply:
ok - will remove the _server_ component then